Intercom error when pressing the back button. Shutting down VM

DecorView@15ef58e[MainActivity]
D/AndroidRuntime( 8635): Shutting down VM
E/AndroidRuntime( 8635): FATAL EXCEPTION: main
E/AndroidRuntime( 8635): Process: com.android.app, PID: 8635
E/AndroidRuntime( 8635): java.lang.RuntimeException: Unable to destroy activity {com.android.app/com.android.app.MainActivity}: kotlin.UninitializedPropertyAccessException: lateinit property unreadConversationCountListener has not been initialized
E/AndroidRuntime( 8635): 	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4589)
E/AndroidRuntime( 8635): 	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4611)
E/AndroidRuntime( 8635): 	at android.app.servertransaction.DestroyActivityItem.execute(DestroyActivityItem.java:39)
E/AndroidRuntime( 8635): 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145)
E/AndroidRuntime( 8635): 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
E/AndroidRuntime( 8635): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1861)
E/AndroidRuntime( 8635): 	at android.os.Handler.dispatchMessage(Handler.java:106)
E/AndroidRuntime( 8635): 	at android.os.Looper.loop(Looper.java:193)
E/AndroidRuntime( 8635): 	at android.app.ActivityThread.main(ActivityThread.java:6819)
E/AndroidRuntime( 8635): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 8635): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:497)
E/AndroidRuntime( 8635): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:912)
E/AndroidRuntime( 8635): Caused by: kotlin.UninitializedPropertyAccessException: lateinit property unreadConversationCountListener has not been initialized
E/AndroidRuntime( 8635): 	at io.maido.intercom.IntercomFlutterPlugin.onDetachedFromEngine(IntercomFlutterPlugin.kt:209)
E/AndroidRuntime( 8635): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.remove(FlutterEngineConnectionRegistry.java:262)
E/AndroidRuntime( 8635): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.remove(FlutterEngineConnectionRegistry.java:270)
E/AndroidRuntime( 8635): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.removeAll(FlutterEngineConnectionRegistry.java:278)
E/AndroidRuntime( 8635): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.destroy(FlutterEngineConnectionRegistry.java:122)
E/AndroidRuntime( 8635): 	at io.flutter.embedding.engine.FlutterEngine.destroy(FlutterEngine.java:365)
E/AndroidRuntime( 8635): 	at io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.onDetach(FlutterActivityAndFragmentDelegate.java:555)
E/AndroidRuntime( 8635): 	at io.flutter.embedding.android.FlutterActivity.release(FlutterActivity.java:589)
E/AndroidRuntime( 8635): 	at io.flutter.embedding.android.FlutterActivity.onDestroy(FlutterActivity.java:610)
E/AndroidRuntime( 8635): 	at android.app.Activity.performDestroy(Activity.java:7395)
E/AndroidRuntime( 8635): 	at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1306)
E/AndroidRuntime( 8635): 	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4574)
E/AndroidRuntime( 8635): 	... 11 more
Lost connection to device.
Exited (sigterm)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,6 @@
+## 2.3.2
+* Fix crash if app is closed before fully initialised
+
 ## 2.3.1
 * Fix Android build issue
 * Updated Android dependencies

--- a/android/src/main/kotlin/io/maido/intercom/IntercomFlutterPlugin.kt
+++ b/android/src/main/kotlin/io/maido/intercom/IntercomFlutterPlugin.kt
@@ -33,7 +33,7 @@ class IntercomFlutterPlugin : FlutterPlugin, MethodCallHandler, EventChannel.Str
   }
 
   private val intercomPushClient = IntercomPushClient()
-  private lateinit var unreadConversationCountListener: UnreadConversationCountListener
+  private var unreadConversationCountListener: UnreadConversationCountListener? = null
 
   override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
     val channel = MethodChannel(flutterPluginBinding.getFlutterEngine().getDartExecutor(), "maido.io/intercom")
@@ -198,15 +198,21 @@ class IntercomFlutterPlugin : FlutterPlugin, MethodCallHandler, EventChannel.Str
 
   override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
     unreadConversationCountListener = UnreadConversationCountListener { count -> events?.success(count) }
-    Intercom.client().addUnreadConversationCountListener(unreadConversationCountListener)
+        .also {
+          Intercom.client().addUnreadConversationCountListener(it)
+        }
   }
 
   override fun onCancel(arguments: Any?) {
-    Intercom.client().removeUnreadConversationCountListener(unreadConversationCountListener)
+    if (unreadConversationCountListener != null) {
+      Intercom.client().removeUnreadConversationCountListener(unreadConversationCountListener)
+    }
   }
 
   override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
-    Intercom.client().removeUnreadConversationCountListener(unreadConversationCountListener)
+    if (unreadConversationCountListener != null) {
+      Intercom.client().removeUnreadConversationCountListener(unreadConversationCountListener)
+    }
   }
 
   override fun onDetachedFromActivity() {

--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -1,7 +1,7 @@
 name: intercom_flutter
 description: Flutter plugin for Intercom integration. Provides in-app messaging
   and help-center Intercom services
-version: 2.3.1
+version: 2.3.2
 homepage: https://github.com/v3rm0n/intercom_flutter
 
 dependencies:

